mlflow/models/evaluation/artifacts.py

```diff
@@ -4,6 +4,8 @@
 
 
 class ImageEvaluationArtifact(EvaluationArtifact):
+    ext = "png"
```

Comment:
Can we make this an `@abstract` `@property` method on `EvaluationArtifact`?

Reply:
But I think there are cases where I want to access the extension without having an instance of the artifact.

Reply:
@staticmethod ?

Reply:
Ohh I see, yeah that works. I was trying to stick with `ext` being a property rather than a function.

Reply:
`@property` makes `ext` appear to be a property. See https://www.programiz.com/python-programming/property for more info.

Reply:
Yes, I am aware of that. This is why I mentioned that I want to access it without having an instance. Because `@property` can only be accessed through an instance. So we have to pick between `@staticmethod` or `@property`.

Reply:
@MarkYHZhang Ah, I missed that. Apologies. How about https://stackoverflow.com/a/7864317/11952869?

Reply:
IMO I think this is a bit overly complicated and as the comment mentioned, not a great use of singleton design pattern. I think we should either stick with what I have right now, or change it to the static method way. Opinions?

Reply:
Personally I'm not a fan of the singleton pattern in general, and I also agree it seems like overkill for this use case.
Ultimately we just want to associate an EvaluationArtifact with its associated file extension with an easy-to-access class property. @staticmethod seems best to me.

Reply:
Updated the code with `@staticmethod` and `@abstractmethod`. I also added `raise NotImplementedError` inside the abstract method. Otherwise, no exception will be thrown when child class provides no implementations to `file_extension()` when called without class instance. Concretely:

```python
from abc import abstractmethod, ABCMeta

class Parent(metaclass=ABCMeta):
    def __init__(self):
        pass
    
    @staticmethod
    @abstractmethod
    def file_extension():
        pass

class Child(Parent):
    def __init__(self):
        pass

Child().file_extension() # will throw TypeError: Can't instantiate abstract class Child with abstract methods ext
Child.file_extension() # will not throw anything, will simply return None.
```